src/citrine/resources/table_config.py

```diff
@@ -349,20 +347,22 @@ def default_for_material(
 
 
         """
+        if scope is not None:
+            warn("\'scope\' is deprecated in default_for_material() and will soon be removed. "
+                 "To specify a custom scope, use a LinkByUID object.", DeprecationWarning)
         if isinstance(material, MaterialRun):
-            if scope is not None:
-                logger.warning(
-                    'Ignoring scope {} since material run object was specified.'.format(scope))
             uid_tup = next(iter(material.uids.items()), None)
             if uid_tup is None:
                 raise ValueError('Material must have a uid to build default table config.')
             scope, uid = uid_tup
```

Comment:
```suggestion
            link = LinkByUID.from_entity(material, CITRINE_SCOPE)
            scope, uid = material.scope, material.id
```
would make more sense, though that fails to catch the `there's no id` scenario.  Looping through internal structs always strikes me as not-great.

Reply:
`LinkByUID.from_entity` has non-obvious behavior in that it fills in an ID if none exists with that scope (thereby missing the the scenario you mention here). We could define a different version of `from_entity` that behaves as we need it to, but we might as well create the desired behavior here.

> Looping through internal structs always strikes me as not-great.

I don't see what the issue is in this case. We just need scope-id pair, and any will work. It's not going to fail in unexpected ways. If there are performance reasons to prefer `CITRINE_SCOPE`,  we could check for that first.

Reply:
> I don't see what the issue is in this case.

The pattern is an issue if we change the implementation of uids away from a simple dict.  Not blocking.